borg/helpers.py

```diff
@@ -146,8 +159,18 @@ def load(cls, repository, key=None, force_tam_not_required=False):
             if not manifest_required and security_required:
                 logger.debug('Manifest is TAM verified and says TAM is *not* required, updating security database...')
                 os.unlink(tam_required_file(repository))
+        manifest.check_repository_compatibility(operation)
         return manifest, key
 
+    def check_repository_compatibility(self, operation):
+        if operation == REPO_OPERATION_NO_CHECK:
+            return
+        if operation not in self.config:
+            return
+        requirements = self.config[operation]
```

Comment:
currently it is structured
`manifest.config[b'write'] = {b'mandatory': [b'unknown-feature']}
`
But i’m open for other layouts. I think there are some features we want to store in the config that don‘t lockout older versions. So i would want to keep some distinction between mandatory and "as supported" features.